jetty-ee10/pom.xml

```diff
@@ -13,12 +13,6 @@
   <packaging>pom</packaging>
 
   <properties>
-    <com.sun.mail.version>2.0.1</com.sun.mail.version>
-    <!-- TODO: Remove these javax.* entries? -->
```

Comment:
The last time I tried removing some of these "old" dependencies it turned out that the testing framework - paxweb - needed them. Let's see how it builds on CI.

Reply:
The OSGI layers and all of the `jetty-ee10/jetty-ee10-osgi/` tree is passing with green so far.

There's 1 failure remaining, a 502 response from the `DistributionTests#testWebAppWithProxyAndJPMS`.
It doesn't seem related to the change in this PR, but rather a CI flaky network issue ATM. 
I just added DEBUG logging to that test case in the last commit (b29d589ebd05123e808f41f3fa3af4d22e73aad9) to see what's going on exactly.